src/sysext/sysext.c

```diff
@@ -309,8 +311,15 @@ static int mount_overlayfs(
                 separator = true;
         }
 
+        /* confexts are nosuid by default, even if they are exec */
+        if (arg_image_class == IMAGE_CONFEXT)
+                flags |= MS_NOSUID;
+
+        if (arg_image_class == IMAGE_CONFEXT && arg_noexec)
+                flags |= MS_NOEXEC;
```

Comment:
i'd prefer if arg_noexec was a tristate, and would always be honoured if configured, even if it makes little sense for sysext...

i.e.:

```c
flags = image_class_info[arg_image_class].default_mount_options;
if (arg_noexec >= 0)
        SET_FLAG(flags, MS_NOEXEC, arg_noexec);
```